File: Fibonacci.f90
program fibonacci
    implicit none
    integer :: i, n
    ! Original: integer :: t1, t2, nextTerm
    ! Fortran's default integer is typically 32-bit (max ~2.1 billion)
    ! This overflows around F(47)
    ! 
    ! Using integer(kind=8) or integer*8 gives 64-bit integers (up to F(93))
    ! For F(100), you would need arbitrary precision libraries, which are
    ! not standard in Fortran and require external packages
    ! 
    ! This version uses integer(kind=8) for extended range up to F(93)
    integer(kind=8) :: t1, t2, nextTerm  ! 64-bit integers for extended range
    
    t1 = 0
    t2 = 1
    
    print *, "Enter the number of terms: "
    read *, n
    
    print *, "Fibonacci Series: "
    print *, "(Note: accurate up to F(93), values beyond will overflow)"
    
    do i = 1, n
        write(*, '(I0, A)', advance='no') t1, ", "
        nextTerm = t1 + t2
        t1 = t2
        t2 = nextTerm
    end do
    
    print *
    
end program fibonacci

File: Fibonacci.pas
program Fibonacci;

var
    i, n: Integer;
    { Original: t1, t2, nextTerm: Integer;
      Pascal's Integer type is typically 32-bit (max ~2.1 billion)
      This overflows around F(47)
      
      Using Int64 (or QWord for unsigned) provides 64-bit integers
      This extends range up to F(93) before overflow
      
      For F(100) and beyond, you would need arbitrary precision libraries
      Free Pascal has some support through external libraries, but it's not standard
      
      This version uses Int64 for extended range up to F(93) }
    t1, t2, nextTerm: Int64;  { 64-bit signed integers for extended range }

begin
    t1 := 0;
    t2 := 1;
    
    Write('Enter the number of terms: ');
    ReadLn(n);
    
    Write('Fibonacci Series: ');
    WriteLn('(Note: accurate up to F(93), values beyond will overflow)');
    
    for i := 1 to n do
    begin
        Write(t1, ', ');
        nextTerm := t1 + t2;
        t1 := t2;
        t2 := nextTerm;
    end;
    
    WriteLn;
end.
